From 80d8bde6b7ca9e20635965329b2365f73ecb6ea1 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:40 -0400
Subject: [usb] wusbcore: convert nested lock to use spin_lock instead of spin_lock_irq

Message-id: <1410887463-20674-149-git-send-email-dzickus@redhat.com>
Patchwork-id: 95497
O-Subject: [RHEL7 PATCH 148/591] usb: wusbcore: convert nested lock to use spin_lock instead of spin_lock_irq
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 7584f2ebc18b6355f21be5fb2f75afbf3f781ce5
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Wed Apr 23 14:32:27 2014 -0500

    usb: wusbcore: convert nested lock to use spin_lock instead of spin_lock_irq

    Nesting a spin_lock_irq/unlock_irq inside a lock that has already
    disabled interrupts will enable interrupts before we are ready when
    spin_unlock_irq is called.  This patch converts the inner lock to use
    spin_lock and spin_unlock instead.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-xfer.c b/drivers/usb/wusbcore/wa-xfer.c
index c783e85..47a67e5 100644
--- a/drivers/usb/wusbcore/wa-xfer.c
+++ b/drivers/usb/wusbcore/wa-xfer.c
@@ -2391,10 +2391,10 @@ error_complete:
   done) {
 
   dev_info(dev, "Control EP stall.  Queue delayed work.\n");
-  spin_lock_irq(&wa->xfer_list_lock);
+  spin_lock(&wa->xfer_list_lock);
   /* move xfer from xfer_list to xfer_errored_list. */
   list_move_tail(&xfer->list_node, &wa->xfer_errored_list);
-  spin_unlock_irq(&wa->xfer_list_lock);
+  spin_unlock(&wa->xfer_list_lock);
   spin_unlock_irqrestore(&xfer->lock, flags);
   queue_work(wusbd, &wa->xfer_error_work);
  } else {
-- 
1.7.1